server/src/db/model/currencyModel.js

```diff
@@ -0,0 +1,49 @@
+const mongoose = require("mongoose");
+const CurrencySchema = require("../schema/currencySchema")
+const CurrencyModel = mongoose.model("currency", CurrencySchema);
+const bacenValidator = require('../../utils/validators/bacenValidator')
+
+
+const currency = {
+
+    find: async function () {
+        return await CurrencyModel.find({}).limit(50)
+    },
+    insert: async function (currencyNames, currencyObj) {
+        const registers = currencyNames.map((key) => ({
+
+            'currencyName': currencyObj[key].code,
+            'high': new Array(currencyObj[key].high),
+            'low': new Array(currencyObj[key].low),
+            'date': new Array(currencyObj[key].create_date)
+        }))
+
+        if (!bacenValidator.validate(registers)) {
+            for (let register in registers) {
+
+
+                const currency = await CurrencyModel.findOne({ currencyName: registers[register].currencyName })
+
+
+
+                if (!currency) {
+                    await CurrencyModel.create(registers[register]);
+                }
+                else {
+
+
+                    await CurrencyModel.updateOne(
+                        { currencyName: registers[register].currencyName },
+                        { $push: { high: registers[register].high, low: registers[register].low, date: registers[register].date } }
+
+
+                    )
+                }
```

Comment:
Fazer queries no banco dentro de um loop é geralmente uma má prática, esse tipo de código gera uma degradação da performance da aplicação.

Tente usar o operador [$in](https://mongoosejs.com/docs/api/query.html#query_Query-in) para trazer todos os `currencies`de uma vês, depois você pode aguardar em um array os item a serem atualizados no banco e usar a função `CurrencyModel.updateMany` para atualizar todos de uma vês.

Reply:
resolvido com 'bulkwrite'